src/Build/Collections/RetrievableEntryHashSet/HashSet.cs

```diff
@@ -275,32 +263,31 @@ public void Clear()
                 Debug.Assert(_buckets != null, "m_buckets was null but m_lastIndex > 0");
 
                 // clear the elements so that the gc can reclaim the references.
-                // clear only up to m_lastIndex for m_slots 
+                // clear only up to m_lastIndex for m_slots
                 Array.Clear(_slots, 0, _lastIndex);
+#if NET6_0_OR_GREATER
+                Array.Clear(_buckets);
```

Comment:
Although I do think this is cleaner, adding the #ifs makes it very not worth it to me.

Reply:
This might have been done because of a slight codegen improvement. (?)

Reply:
The reason I used it was because CORE used it. When I checked the source code for it, it was a lot more simple. So it should run faster.